TeachingRecordSystem/src/TeachingRecordSystem.Core/Jobs/InductionStatusUpdatedSupportJob.cs

```diff
@@ -0,0 +1,83 @@
+using System.Text.Json;
+using Microsoft.Extensions.Options;
+using TeachingRecordSystem.Core.DataStore.Postgres;
+using TeachingRecordSystem.Core.DataStore.Postgres.Models;
+using TeachingRecordSystem.Core.Dqt;
+using TeachingRecordSystem.Core.Dqt.Queries;
+using TeachingRecordSystem.Core.Jobs.Scheduling;
+
+namespace TeachingRecordSystem.Core.Jobs;
+
+public class InductionStatusUpdatedSupportJob(TrsDbContext dbContext, ICrmQueryDispatcher crmQueryDispatcher, IClock clock, IOptions<InductionStatusUpdatedSupportJobOptions> inductionStatusUpdatedSupportJobOptions)
+{
+    public const string JobSchedule = "0 3 * * *";  //3AM every day
+    public const string LastRunDateKey = "LastRunDate";
+
+    public async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        var item = await dbContext.JobMetadata.SingleOrDefaultAsync(i => i.JobName == nameof(InductionStatusUpdatedSupportJob));
+        var lastRunDate = inductionStatusUpdatedSupportJobOptions.Value.InitialLastRunDate;
+
+        //if the job has ran before - the lastrundate from metadata is used.
+        if (item != null)
+        {
+            if (item.Metadata.TryGetValue(LastRunDateKey, out var obj) &&
+                obj is JsonElement jsonElement &&
+                jsonElement.ValueKind == JsonValueKind.String)
+            {
+                var dateString = jsonElement.GetString();
+                if (DateTime.TryParse(dateString, out DateTime dbLastRunDate))
+                {
+                    lastRunDate = dbLastRunDate;
+                }
+            }
+        }
+
+        var events = await dbContext.Events
+           .Where(e => e.EventName == nameof(PersonInductionUpdatedEvent))
+           .ToListAsync();
+
+        var groupedEvents = events
```

Comment:
Is there a better name for this? Doesn't look like there's any grouping going on.